gravitee-am-service/src/main/java/io/gravitee/am/service/validators/resource/http/HttpResourceValidatorImpl.java

```diff
@@ -0,0 +1,80 @@
+/**
+ * Copyright (C) 2015 The Gravitee team (http://gravitee.io)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.gravitee.am.service.validators.resource.http;
+
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.gravitee.am.service.exception.InvalidParameterException;
+import io.gravitee.am.service.validators.resource.ResourceValidator;
+import org.springframework.stereotype.Component;
+
+import java.util.Optional;
+
+import static java.util.Optional.ofNullable;
+
+/**
+ * @author Eric LELEU (eric.leleu at graviteesource.com)
+ * @author GraviteeSource Team
+ */
+@Component
+public class HttpResourceValidatorImpl implements HttpResourceValidator {
+
+    public static final String UNEXPECTED_MESSAGE = "An unexpected error has occurred while trying to validate resource [%s]";
+    public static final String FIELD_AUTH = "auth";
+    public static final String FIELD_TYPE = "type";
+    public static final String OAUTH2 = "oauth2";
+    public static final String FIELD_ENDPOINT = "endpoint";
+    public static final String FIELD_CLIENTID = "clientId";
+    public static final String FIELD_CLIENT_SECRET = "clientSecret";
+    private final ObjectMapper objectMapper;
+    public static final String HTTP_AM_RESOURCE = "http-am-resource";
+    public static final String INVALID_OAUTH_CONF = "OAuth config is missing or invalid";
+
```

Comment:
**wondering** I understand that static fields are public because they are used in unit tests, but I'm wondering if we shouldn't use literals there. If you changed value in HTTP_AM_RESOURCE, the test would pass?

Reply:
I changed the visibility